products/llbuildSwift/BuildDBBindings.swift

```diff
@@ -0,0 +1,172 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright 2015-2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+// This file contains Swift bindings for the llbuild C API.
+
+#if os(macOS)
+import Darwin.C
+#elseif os(Windows)
+import MSVCRT
+import WinSDK
+#else
+import Glibc
+#endif
+
+import Foundation
+
+// We don't need this import if we're building
+// this file as part of the llbuild framework.
+#if !LLBUILD_FRAMEWORK
+import llbuild
+#endif
+
+public typealias KeyID = UInt64
+public typealias KeyType = String
+
+/// Defines the result of a call to fetch all keys from the database
+/// Wraps calls to the llbuild database, but all results are fetched and available with this result
+public class BuildDBKeysResult {
```

Comment:
Let's land the C API, then we can decide what the Swift API should look like.

Reply:
Opened a new PR [here](https://github.com/apple/swift-llbuild/pull/485) for the C interface. Since I renamed the remote branch, Github automatically closed the PR but I think that's fine. I'll open another for the Swift interface once the C one is merged.